Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -1,173 +1,160 @@
 /-
-Copyright (c) 2020 Simon Hudon. All rights reserved.
+Copyright (c) 2023 Harun Khan. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Simon Hudon
+Authors: Harun Khan
 -/
-import Mathlib.Data.Bitvec.Defs
-import Mathlib.Data.Fin.Basic
-import Mathlib.Tactic.NormNum
 
-#align_import data.bitvec.basic from "leanprover-community/mathlib"@"008af8bb14b3ebef7e04ec3b0d63b947dee4d26a"
+import Mathlib.Data.Bitvec.Defs
 
 /-!
 # Basic Theorems About Bitvectors
 
-This file contains theorems about bitvectors and their coercions to
-`Nat` and `Fin`.
+This file contains theorems about bitvectors.
 -/
-namespace Bitvec
+
+namespace Std.BitVec
 
 open Nat
 
-theorem bitsToNat_toList {n : ℕ} (x : Bitvec n) : Bitvec.toNat x = bitsToNat (Vector.toList x) :=
+variable {w v : Nat}
+
+/-! We add simp-lemmas that rewrite bitvector operations into the equivalent notation -/
+@[simp] lemma append_eq (x : BitVec w) (y : BitVec v)   : BitVec.append x y = x ++ y        := rfl
+@[simp] lemma shiftLeft_eq (x : BitVec w) (n: Nat)      : BitVec.shiftLeft x n = x <<< n    := rfl
+@[simp] lemma ushiftRight_eq (x : BitVec w) (n: Nat)    : BitVec.ushiftRight x n = x >>> n  := rfl
+@[simp] lemma not_eq (x : BitVec w)                     : BitVec.not x = ~~~x               := rfl
+@[simp] lemma and_eq (x y : BitVec w)                   : BitVec.and x y = x &&& y          := rfl
+@[simp] lemma or_eq (x y : BitVec w)                    : BitVec.or x y = x ||| y           := rfl
+@[simp] lemma xor_eq (x y : BitVec w)                   : BitVec.xor x y = x ^^^ y          := rfl
+@[simp] lemma neg_eq (x : BitVec w)                     : BitVec.neg x = -x                 := rfl
+@[simp] lemma add_eq (x y : BitVec w)                   : BitVec.add x y = x + y            := rfl
+@[simp] lemma sub_eq (x y : BitVec w)                   : BitVec.sub x y = x - y            := rfl
+@[simp] lemma mul_eq (x y : BitVec w)                   : BitVec.mul x y = x * y            := rfl
```

Comment:
I think you could PR these ones straight to Std, since they're fundamentally about deciding what's the API and what's an implementation detail.

Reply:
leanprover/std4#357